### PR TITLE
Remove dead code, do struct alignment, simplify code

### PR DESCRIPTION
**Pull request by wendigo**
I've passed code through gometaliner and made some changes: aligned structs, removed dead/unused code, simplified some statements.

### CODE REVIEW

File: caddyhttp/fastcgi/fcgiclient_test.go [hunk 1]
```diff
@@ -155,7 +155,7 @@ func sendFcgi(reqType int, fcgiParams map[string]string, data []byte, posts map[
 	fcgi.Close()
 	time.Sleep(1 * time.Second)
 
-	if bytes.Index(content, []byte("FAILED")) >= 0 {
+	if bytes.Contains(content, []byte("FAILED")) {
```
Comment (wendigo) [inline caddyhttp/fastcgi/fcgiclient_test.go]: bytes has Contains method as well

File: caddyhttp/fastcgi/fcgiclient.go [hunk 2]
```diff
@@ -261,29 +260,6 @@ func (c *FCGIClient) writePairs(recType uint8, pairs map[string]string) error {
 	return nil
 }
 
-func readSize(s []byte) (uint32, int) {
```
Comment (wendigo) [inline caddyhttp/fastcgi/fcgiclient.go]: These two methods were unused

File: caddyhttp/fastcgi/fcgiclient.go [hunk 3]
```diff
@@ -44,7 +44,6 @@ const FCGINullRequestID uint8 = 0
 
 // FCGIKeepConn describes keep connection mode.
 const FCGIKeepConn uint8 = 1
-const doubleCRLF = "\r\n\r\n"
```
Comment (wendigo) [inline caddyhttp/fastcgi/fcgiclient.go]: This was unused across the codebase

File: caddyfile/parse.go [hunk 4]
```diff
@@ -76,6 +70,7 @@ func (p *parser) begin() error {
 	}
 
 	err := p.addresses()
+
```
Comment (mholt) [inline caddyfile/parse.go]: I'm going to be pedantic here and suggest that this empty line shouldn't be here, as the error check on the next line is directly associated with this line. ;)

Anyway, I won't make this a stopper for the PR if it's the only change I would want.

File: caddytls/config.go [hunk 5]
```diff
@@ -418,27 +418,6 @@ var supportedCiphersMap = map[string]uint16{
 	"RSA-3DES-EDE-CBC-SHA":          tls.TLS_RSA_WITH_3DES_EDE_CBC_SHA,
 }
 
-// List of supported cipher suites in descending order of preference.
-// Ordering is very important! Getting the wrong order will break
-// mainstream clients, especially with HTTP/2.
-//
-// Note that TLS_FALLBACK_SCSV is not in this list since it is always
-// added manually.
-var supportedCiphers = []uint16{
```
Comment (mholt) [inline caddytls/config.go]: Wow, I guess I must've stopped using this with the 0.9 rewrite and not noticed it.

Makes sense though, I don't see why we would need this now. The order is still important, but on the defaultCiphers slice (below) -- I'll have to update the godoc comments (aside from this) - thanks!

**Pull request merged**